### PR TITLE
[#162412][Daily Rate] Schedule rules updates

### PR DESCRIPTION
**Pull request by joaquinco**
## Notes

- Update schedule rules checks for daily booking instrument's reservations to just check availability of the start date
- Add tests for create and move reservations

### CODE REVIEW

File: spec/models/reservations_validations_spec.rb [hunk 1]
```diff
@@ -80,4 +79,84 @@
...
+  describe "schedule rules" do
+    context "daily booking instrument" do
+      subject(:reservation) { build :setup_reservation, product: }
+      let(:product) { create :setup_instrument, :daily_booking }
+
+      before do
+        product.update_attribute(
+          :pricing_mode, Instrument::Pricing::SCHEDULE_DAILY
+        )
+      end
```
Comment (LeticiaErrandonea) [inline spec/models/reservations_validations_spec.rb:92]: Shouldn't this be done by the `daily_booking` trait?

File: spec/models/reservations_validations_spec.rb [hunk 2]
```diff
@@ -80,4 +79,84 @@
...
+        context "start at is cover and so is end at" do
+          before do
+            create(
+              :schedule_rule,
+              :all_day,
+              product:,
+            )
+          end
+
+          it { expect(product.schedule_rules.cover?(start_at, end_at)).to be true }
```
Comment (LeticiaErrandonea) [inline spec/models/reservations_validations_spec.rb:156]: Should we also add?
```rb
it { expect(product.schedule_rules.cover_time?(start_at)).to be true }
```

**Review by LeticiaErrandonea — APPROVED** [review]
Left a couple of comments/questions but LGTM

File: spec/system/editing_a_reservation_spec.rb [hunk 3]
```diff
@@ -91,4 +91,152 @@
...
+          reserve_end_at: new_reserve_start_at + 1.day,
+          product: instrument
+        )
+
+        expect(invalid_reservation).to_not be_valid
+        expect(invalid_reservation.errors).to be_added(:base, error_key)
+      end
+    end
+
+    describe "when unavailable due to schedule rule" do
```
Comment (diebas) [inline spec/system/editing_a_reservation_spec.rb:156]: I think context block fits better in this cases

https://dev.to/vasily/explaining-rspec-describe-context-in-2-minutes-264h

File: spec/system/editing_a_reservation_spec.rb [hunk 4]
```diff
@@ -91,4 +91,152 @@
...
+        reservation.product.schedule_rules.first.then do |schedule_rule|
+          wday = unavailable_period_start.wday
+          wday_name = Date::ABBR_DAYNAMES[wday].downcase
+          schedule_rule.update("on_#{wday_name}" => false)
+        end
+      end
+
+      include_examples "unavailable due to reason", :no_schedule_rule
+
+      describe "when the reservation last 1 day" do
```
Comment (diebas) [inline spec/system/editing_a_reservation_spec.rb:168]: same here and the below blocks

**Review by diebas — APPROVED** [review]
Great job! just a couple of  minor comments

File: spec/models/reservations_validations_spec.rb [hunk 5]
```diff
@@ -80,4 +79,84 @@
...
+
+  describe "schedule rules" do
+    context "daily booking instrument" do
+      subject(:reservation) { build :setup_reservation, product: }
+      let(:product) { create :setup_instrument, :daily_booking }
+
+      before do
+        product.update_attribute(
+          :pricing_mode, Instrument::Pricing::SCHEDULE_DAILY
+        )
```
[the remark this answers is not in the document]
Comment (joaquinco) [reply spec/models/reservations_validations_spec.rb:91]: indeed

**Pull request merged**